astropy/modeling/fitting.py

```diff
@@ -415,19 +415,23 @@ def __call__(self, model, x, y, z=None, weights=None, **kwargs):
 
         fitted_model = self.fitter(model, x, y, z, weights, **kwargs)
         if z is None:
+            fitted_model = self.fitter(fitted_model, x, y, **kwargs)
             filtered_data = y
             for n in range(self.niter):
-                filtered_data = self.outlier_func(filtered_data,
+                filtered_data = self.outlier_func(filtered_data - fitted_model(x),
                                                   **self.outlier_kwargs)
+                filtered_data += fitted_model(x)
```

Comment:
I think ``filtered_data`` is meant to be a subset of the original data provided by the user (data set - outliers set).
Looks like this line breaks that concept?